src/codeu/chat/server/Server.java

```diff
@@ -80,24 +66,27 @@ public void onMessage(InputStream in, OutputStream out) throws IOException {
         final Uuid conversation = Uuid.SERIALIZER.read(in);
         final String content = Serializers.STRING.read(in);
 
+        checkBuffer();
```

Comment:
don't checkBuffer() here, see my comment near the checkBuffer() function for details.